compiler/qsc_frontend/src/incremental/tests.rs

```diff
@@ -469,6 +470,176 @@ fn continue_after_lower_error() {
     "#]].assert_debug_eq(&errors);
 }
 
+#[test]
+fn reexports_complex() {
+    let mut store = PackageStore::new(compile::core());
+
+    let package_a = SourceMap::new(
+        [(
+            "PackageA.qs".into(),
+            indoc! {"
+                operation Foo(x: Int, y: Bool) : Int {
+                    x
+                }
+                export Foo;
+            "}
+            .into(),
+        )],
+        None,
+    );
+
+    let package_a = compile::compile(
+        &store,
+        &[],
+        package_a,
+        TargetCapabilityFlags::all(),
+        LanguageFeatures::default(),
+    );
+    assert!(package_a.errors.is_empty(), "{:#?}", package_a.errors);
+
+    let package_a = store.insert(package_a);
+
+    let package_b = SourceMap::new(
+        [(
+            "PackageB.qs".into(),
+            indoc! {"
+                import A.PackageA.Foo;
```

Comment:
nit: IMO this test is doing a lot. I think it could be broken down into multiple tests:
- Importing with alias.
- Re-exporting with alias.

It also seems like importing with no alias is a missing test case in this file.

A lot of the code for these test cases would be shared so it would also be good to create a few utility functions to make the code of each test more concise.